test-app/build.gradle.kts

```diff
@@ -100,6 +100,9 @@ dependencies {
     // AM NOTE: needs to stay this version for now (June 24,2020)
     implementation("org.jsoup:jsoup:1.14.3")
 
+    implementation("androidx.media2:media2-session:1.2.0")
+    implementation("androidx.media2:media2-player:1.2.0")
```

Comment:
You can use `api` instead of `implementation` in `navigator-media/build.gradle` to make the `media2` types automatically visible in the Test App when importing `readium-navigator-media`, without requiring to import them explicitly.

Reply:
Yes, I can. Should I? It might be a better practice to let modules be honest about which modules they depend on.

Reply:
IMO, a good rule of thumb is that if you expose a type from a nested dependency in your public surface API, then effectively the nested dependency is now part of your own public API and you _should_ use `api` (might be why they chose this keyword).

However, I don't have a strong opinion for this case as the type of `media2` that are exposed are not required to be used in the test app for the navigator to work, so an explicit opt-in seems fine.

Anyways, I tried commenting these two `implementation` statements and even without the use of `api` the test app still builds and runs fine (I cleared all the cache). This is weird no?

Reply:
Ha, I think it's because we use `api` for ExoPlayer in the `navigator` module: 
https://github.com/readium/kotlin-toolkit/blob/72fa156d85173c186bb0f2aae020be2122c6abc2/readium/navigator/build.gradle.kts#L90-L94

And ExoPlayer itself is using `api` for the `media2` dependency (per the rule of thumb above), so transitively this is accessible in the Test App.

https://github.com/google/ExoPlayer/blob/029a2b27cbdc27cf9d51d4a73ebeb503968849f6/extensions/media2/build.gradle#L20

For now you can leave it like this and we'll adjust when removing the older media1 navigator.

Reply:
Ok